src/hw4/stack.c

```diff
@@ -0,0 +1,38 @@
+#include "stack.h"
+#include <stdlib.h>
+
+struct Stack newStack(void) {
+  struct Stack stack = {.head = NULL};
+  return stack;
+}
+
+void push(struct Stack *stack, int value) {
+  struct StackNode *node = malloc(sizeof(struct StackNode));
```

Comment:
Здесь надо бы сделать проверку того, действительно ли выделилась память, и вернуть код ошибки, если не получилось.
Есть два решения:
- сделать так чтобы либо функция возвращала bool
- или принимала указатель на код ошибки `int *errorCode` как параметр
```suggestion
  struct StackNode *node = malloc(sizeof(struct StackNode));
  if (node == NULL) {
    return false;
    // или
    *errorCode = 1;
  }
```